src/codec.rs

```diff
@@ -782,36 +794,67 @@ impl<T: Encode> Encode for [T] {
 	}
 }
 
-/// Read an `u8` vector from the given input.
-pub(crate) fn read_vec_u8<I: Input>(input: &mut I, len: usize) -> Result<Vec<u8>, Error> {
+/// Create a `Vec<T>` by casting directly from a buffer of read `u8`s
+///
+/// The encoding of `T` must be equal to its binary representation, and size of `T` must be less or
+/// equal to [`MAX_PREALLOCATION`].
+pub(crate) fn read_vec_from_u8s<I, T>(input: &mut I, items_len: usize) -> Result<Vec<T>, Error>
```

Comment:
yes I think we can do it, we can probably improve some code in BE, but actually I wanted to do as least effort as possible to support BE platform for now, as I don't know any ppl who need it.

I also introduced reverse_endian in bitvec only because it was needed as `T` didn't even bound `Decode`. FYI in the follow-up PR I remove reverse_endian.

Reply:
Is there a follow up pr? (or in the future)

Reply:
in the future, but should be soon ready

Reply:
it is here, https://github.com/paritytech/parity-scale-codec/pull/236

Thus the removal of reverse endian, (and probably bitvec performance regression on BE)